File: src/util/Encrypt.java
package util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import static java.security.MessageDigest.getInstance;

public class Encrypt {
    public static byte[] SHA1(String password, String salt) {
        return SHA1(password.getBytes(), salt.getBytes());
    }

    public static byte[] SHA1(byte[] password, byte[] salt) {
        try {
            MessageDigest md = getInstance("SHA-1");
            md.update(salt);
            return md.digest(password);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/security/Encrypt.kt
package security

import java.security.MessageDigest.getInstance
import java.security.NoSuchAlgorithmException

internal object Encrypt {
    fun sha1(password: String, salt: String): ByteArray {
        return sha1(password.toByteArray(), salt.toByteArray())
    }

    private fun sha1(password: ByteArray, salt: ByteArray): ByteArray {
        return try {
            val md = getInstance("SHA-1")
            md.update(salt)
            md.digest(password)
        } catch (e: NoSuchAlgorithmException) {
            throw RuntimeException(e)
        }
    }
}
